addons/account_payment/models/account_payment.py

```diff
@@ -204,7 +207,7 @@ def _prepare_payment_transaction_vals(self, **extra_create_values):
         self.ensure_one()
         return {
             'provider_id': self.payment_token_id.provider_id.id,
-            'reference': self.ref,
+            'reference': self.env['payment.transaction']._compute_reference(self.payment_token_id.provider_id.code, self.ref),
```

Comment:
Breaking the line for readability on smaller displays + always pass kwarg values as kwargs to avoid bad surprises with refactors.
```suggestion
            'reference': self.env['payment.transaction']._compute_reference(
                self.payment_token_id.provider_id.code, prefix=self.ref
            ),
```

Reply:
Done